DataManagementSystem/Agent/RequestOperations/MoveReplica.py

```diff
@@ -0,0 +1,447 @@
+""" MoveReplica operation handler
+
+    This handler moves replicas from source SEs to target SEs. Replicas are first replicated to target SEs and then removed from the source SEs
+"""
+
+__RCSID__ = "$Id $"
+
+# # imports
+import re, os, string
+# # from DIRAC
+from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC.FrameworkSystem.Client.MonitoringClient import gMonitor
+from DIRAC.Core.Utilities.Adler import compareAdler
+
+from DIRAC.DataManagementSystem.Client.DataManager                                import DataManager
+from DIRAC.DataManagementSystem.Agent.RequestOperations.DMSRequestOperationsBase  import DMSRequestOperationsBase
+
+from DIRAC.Resources.Storage.StorageElement                                       import StorageElement
+from DIRAC.Resources.Catalog.FileCatalog                                          import FileCatalog
+
+def filterReplicas( opFile, logger = None, dataManager = None ):
+  """ filter out banned/invalid source SEs """
+
+  if logger is None:
+    logger = gLogger
+  if dataManager is None:
+    dataManager = DataManager()
+
+  log = logger.getSubLogger( "filterReplicas" )
+  ret = { "Valid" : [], "NoMetadata" : [], "Bad" : [], 'NoReplicas':[], 'NoPFN':[] }
+  replicas = dataManager.getActiveReplicas( opFile.LFN )
+
+  if not replicas["OK"]:
+    log.error( 'Failed to get active replicas', replicas["Message"] )
+    return replicas
+  reNotExists = re.compile( r".*such file.*" )
+  replicas = replicas["Value"]
+  failed = replicas["Failed"].get( opFile.LFN , "" )
+  if reNotExists.match( failed.lower() ):
+    opFile.Status = "Failed"
+    opFile.Error = failed
+    return S_ERROR( failed )
+
+  replicas = replicas["Successful"].get( opFile.LFN, {} )
+  noReplicas = False
+  if not replicas:
+    allReplicas = dataManager.getReplicas( opFile.LFN )
+    if allReplicas['OK']:
+      allReplicas = allReplicas['Value']['Successful'].get( opFile.LFN, {} )
+      if not allReplicas:
+        ret['NoReplicas'].append( None )
+        noReplicas = True
+      else:
+        # We try inactive replicas to see if maybe the file doesn't exist at all
+        replicas = allReplicas
+      log.warn( "File has no%s replica in File Catalog" % ( '' if noReplicas else ' active' ), opFile.LFN )
+    else:
+      return allReplicas
+
+  if not opFile.Checksum:
```

Comment:
For these types of checks please look at the `ConsistencyInspector` class (in `DataManagemen.Client`)